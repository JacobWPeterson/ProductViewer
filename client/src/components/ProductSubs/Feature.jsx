import React from 'react';
import styled from 'styled-components';

const ListItem = styled.li`
  &:before { content: 'âœ“';
    color: #80ccc4;
    padding-right: .5rem; };
  font-weight: 300;
  color: #424242;
  line-height: 2rem;
`;

const Feature = ({ feature }) => {
  if (feature.value) {
    return (
      <ListItem>{`${feature.feature}: ${feature.value}`}</ListItem>
    );
  }
  return (
    <ListItem>{feature.feature}</ListItem>
  );
};

export default Feature;
