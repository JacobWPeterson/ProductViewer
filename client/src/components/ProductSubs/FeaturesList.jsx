import React from 'react';
import styled from 'styled-components';
import Feature from './Feature.jsx';

const List = styled.ul`
  list-style: none;
  flex-basis: 33%;
  padding-left: 1.5vw;
`;

const FeaturesList = ({ id, features }) => (
  <List>
    {features.map((feature) => <Feature key={`${id} + ${feature.feature}`} feature={feature} />)}
  </List>
);

export default FeaturesList;
