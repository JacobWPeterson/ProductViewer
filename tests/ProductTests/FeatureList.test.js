/* eslint-disable no-undef */
/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { mount } from 'enzyme';

import FeaturesList from '../../client/src/components/ProductSubs/FeaturesList.jsx';
import Feature from '../../client/src/components/ProductSubs/Feature.jsx';
import ProductTestData from '../ProductTestData.js';

const props = {
  features: ProductTestData.productDetails.features,
};

describe('FeaturesList', () => {
  const FeaturesListComponent = mount(<FeaturesList {...props} debug />);

  it('Should render correctly', () => {
    expect(FeaturesListComponent).toMatchSnapshot();
  });

  it('renders the feature list', () => {
    expect(FeaturesListComponent.find(Feature).length).toEqual(3);
  });
});
