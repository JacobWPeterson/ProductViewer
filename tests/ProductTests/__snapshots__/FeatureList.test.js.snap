// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturesList Should render correctly 1`] = `
<FeaturesList
  debug={true}
  features={
    Array [
      Object {
        "feature": "Fabric",
        "value": "100% Cotton",
      },
      Object {
        "feature": "Cut",
        "value": "Skinny",
      },
      Object {
        "feature": "Material",
        "value": null,
      },
    ]
  }
>
  <FeaturesList__List>
    <ul
      className="FeaturesList__List-btq5gp-0 gsMclF"
    >
      <Feature
        feature={
          Object {
            "feature": "Fabric",
            "value": "100% Cotton",
          }
        }
        key="undefined + Fabric"
      >
        <Feature__ListItem>
          <li
            className="Feature__ListItem-sc-2j09xg-0 kZXsUK"
          >
            Fabric: 100% Cotton
          </li>
        </Feature__ListItem>
      </Feature>
      <Feature
        feature={
          Object {
            "feature": "Cut",
            "value": "Skinny",
          }
        }
        key="undefined + Cut"
      >
        <Feature__ListItem>
          <li
            className="Feature__ListItem-sc-2j09xg-0 kZXsUK"
          >
            Cut: Skinny
          </li>
        </Feature__ListItem>
      </Feature>
      <Feature
        feature={
          Object {
            "feature": "Material",
            "value": null,
          }
        }
        key="undefined + Material"
      >
        <Feature__ListItem>
          <li
            className="Feature__ListItem-sc-2j09xg-0 kZXsUK"
          >
            Material
          </li>
        </Feature__ListItem>
      </Feature>
    </ul>
  </FeaturesList__List>
</FeaturesList>
`;
